c/cert/src/rules/DCL30-C/AppropriateStorageDurationsFunctionReturn.ql

```diff
@@ -0,0 +1,50 @@
+/**
+ * @id c/cert/appropriate-storage-durations-function-return
+ * @name DCL30-C: Declare objects with appropriate storage durations
+ * @description When pointers to local variables are returned by a function it can lead to referring
+ *              to objects outside of their lifetime, which is undefined behaviour.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/cert/id/dcl30-c
+ *       correctness
+ *       external/cert/obligation/rule
+ */
+
+import cpp
+import codingstandards.c.cert
+import codingstandards.cpp.FunctionParameter
+import semmle.code.cpp.dataflow.DataFlow
+
+class Source extends StackVariable {
+  Source() { not this instanceof FunctionParameter }
```

Comment:
I don't think `FunctionParameter` is needed here - you want to exclude all parameters, not just those from definitions or instantiated templates.
```suggestion
  Source() { not this instanceof Parameter }
```

Reply:
agree, addressed!